code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -83,26 +83,21 @@
 
 //////////////HOSTILE MOB TARGETTING AND AGGRESSION////////////
 
-
 /mob/living/simple_animal/hostile/proc/ListTargets()//Step 1, find out what we can see
 	. = list()
 	if(!search_objects)
 		var/list/Mobs = hearers(vision_range, targets_from) - src //Remove self, so we don't suicide
```

Comment:
there is much more wrong with this proc then what oranges pointed out, i decided against changing anything function wise about the proc.

This could be changed, to a type cached view or hear() search, merged with the mech/turret stuff, and get rid of can_hear() to be much much faster (most likely, i never profiled it)

The only issue is that can_see()/can_hear() and view()/hear() calculates things differently, and that would mark a change in functionality, so i didn't include it here as i have no idea why it does things the way it does.